File: BehaviorPattern/Observer/Python/Concretes/Zombie.py
from Interfaces import Subject, Observer

class Zombie(Subject): 
    """
        Class Zombie implements the interface Subject
    """
    def __init__(self):
        super().__init__()
        self.observers = []
        self.action = None

    def addObserver(self , observer): 
        self.observers.append(observer)
    
    def removeObserver(self , observer): 
        self.observers.remove(observer)
    
    def notifyObservers(self):
        for observer in self.observers:
            observer.update(self.action)
        
    def setAction(self, action):
        self.action = action
        self.notifyObservers()
    
    def getAction(self) -> str:
        return self.action
    

File: BehaviorPattern/Observer/Java/Concretes/Zombie.java
package Concretes;

import java.util.ArrayList;
import java.util.List;

import Interfaces.Subject;
import Interfaces.Observer;

public class Zombie implements Subject {
    private List<Observer> observers = new ArrayList<>();
    private String action;

    @Override
    public void addObserver(Observer observer) {
        observers.add(observer);
    }
    @Override
    public void removeObserver(Observer observer) {
        observers.remove(observer);
    }
    @Override
    public void notifyObservers() {
        for (Observer observer : observers) {
            observer.update(action);
        }
    }

    public void setAction(String action) {
        this.action = action;
        notifyObservers();
    }

    public String getAction() {
        return action;
    }
}
